File: Scripts/server/inheritance-multiple/ruby/inheritance-multiple.rb
module Camera
  # define methods here
  def shootPicture()
    puts "Say cheese\n"
  end
end

class MobilePhone
  # define methods here
  def self.inherited(subclass)
    puts "new subclass of #{self}: #{subclass}"
  end

  def call()
    puts "Pick a call\n"
  end
end

class CameraPhone < MobilePhone
  include Camera
  # define methods here
end


for i in 0..1000000
  my_class = CameraPhone.new #allocates an object and calls it's initialize method, then returns it.
  my_class.call
  my_class.shootPicture  
end

File: Scripts/server/inheritance-multiple/c++/inheritance-multiple.cpp
#include <iostream>

class Camera
{
  // ...
	public:
		void shootPicture() {
			std::cout<<"Say cheese\n";
		}	
};

class MobilePhone
{
  // ...
	public:
		void call() {
			std::cout<<"Pick a phone call\n";
		}
};

class CameraPhone:
  public Camera,
  public MobilePhone
{
 // ...
};

int main() {

for ( int i = 0; i < 1000000; ++i ) {
	CameraPhone ziaomi;
	ziaomi.call();
	ziaomi.shootPicture();
}
return 0;
}
